### PR TITLE
Removed `hash_and_equals` from the default set of lints.

### PR DESCRIPTION
**Pull request by kevmoo**
There are valid reasons to not follow this rule.

### CODE REVIEW

**Comment by matanlurey** [comment]
I honestly wouldn't use any lints to determine quality score.

We don't have any other tooling that tells you to use them, and internally we've banned quite a bit.

**Comment by kevmoo** [comment]
> I honestly wouldn't use any lints to determine quality score.

Hrm. The set we have here are pretty safe, I think. What do you think, @isoos ?

**Comment by matanlurey** [comment]
I'm not sure it's fair to punish people who are 100% strong-mode clean, and don't know about another package's rules to determine package health. But I'll let you decide if we should do it or not.

**Comment by isoos** [comment]
1) lint-related code checks may decide whether you get 97 or 96, but their impact is roughly that. More precisely: if you get roughly ~1000 lines, and you have one INFO lint failure, your score will be 999/1000, which on the UI will be 100.

2) Lints are useful signals that can detect problems. If you know better, you can always do in-line comment exclude, and make the lint message go away, and `pana` won't see it (or penalize it) either. It is about awareness: we see a potentially harmful pattern, and the code author may or may not indicate whether they are aware of it (either by fixing it or by the comment).

3) You can get equals and hashCode generated for free in IntelliJ. They are linked so much that there isn't a separate menu item to generate equals only or the other. Unless you have a very complex and elaborate case, I cannot imagine why you'd have one, and not the other. In that case, add the comment and you'll be fine.

4) I'm pretty sure that sooner or later everybody will raise an objection for everything that makes their code look a bit less than the rest. At that point we can just remove the code health metric, because it becomes meaningless.

**Comment by isoos** [comment]
TL;DR: no, I wouldn't remove this lint rule.

**Comment by isoos** [comment]
Btw. stagehand has it in their templates too: https://github.com/dart-lang/stagehand/blob/master/templates/package-simple/analysis_options.yaml

**Comment by kevmoo** [comment]
Hrm...okay

**Comment by matanlurey** [comment]
I'm still totally against including any lints in the health score, but I also don't care to argue too much about this. If I see packages having lower scores for not adhering to lints they don't exist, I'll raise this issue again.

**Comment by kevmoo** [comment]
> I'm still totally against including any lints in the health score, but I also don't care to argue too much about this. If I see packages having lower scores for not adhering to lints they don't exist, I'll raise this issue again.

Understood. We need to do some coordination here w/ stagehand, etc.
